python_tests/mpi_test.py

```diff
@@ -0,0 +1,61 @@
+# run it with
+# NVFUSER_DUMP=fusion_ir mpirun --allow-run-as-root -n 2 python mpi_nvfuser.py
+
+import torch
+import nvfuser
+from nvfuser.pytorch_utils import torch_dtype_to_nvfuser_dtype
+from torch import Tensor
+from typing import Tuple
+
+from nvfuser import FusionDefinition, DataType
+
+import os
+
+rank = int(os.environ["OMPI_COMM_WORLD_LOCAL_RANK"])
+
+torch.cuda.set_device(rank)
+
+### Inputs
+inputs = [
+    torch.randn(2, 4, device="cuda")[rank:rank+1, ...],
+]
+
+class MultiDeviceModel(FusionDefinition):
+
+    def definition(self):
+        # dynamic shape isn't working? at least I'm getting asserts
+        #self.t0 = self.from_pytorch(inputs[0])
+
+        self.t0 = self.define_tensor((2, 4), (False, False), dtype=DataType.Float)
+
+        # looks like I cannot have scalar inputs to any expression, hitting an assert
```

Comment:
We only allow TensorViews for now. This restriction was just a simplifying assumption, if needed we can add the support for this